lib/test.js

```diff
@@ -22,6 +22,7 @@ function Test(title, fn) {
   Runnable.call(this, title, fn);
   this.pending = !fn;
   this.type = 'test';
+  this.body = (fn || '').toString();
```

Comment:
Shouldn't the initialization of `this.body` belong to `Runnable`'s constructor rather than `Test`?

Right now, if you get an error in a hook, Mocha will bomb, because the `Hook` object that it tries to process [here](https://github.com/mochajs/mocha/pull/2068/files#diff-998e3fb668a2094eedccab3e56199e4fR203) does not have a `body` field. Moving the creation of `body` to `Runnable` would make both `Hook` and `Test` get a `body` field.

The issue I'm talking about here is fully reproducible with the code provided in [this comment](https://github.com/mochajs/mocha/issues/2067#issuecomment-181430571). I also am able to reproduce it independently in a test suite of mine (which is why I found the comment and am commenting here).